### DISCUSSION TITLE
Friendly name issue with 8 relay board

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#19956] [Support request]
Hi all,

Alright, I booted up this ATHOM relay x8 (8 relay), already installed with an old version of Tasmota.  Then connected Wi-Fi the usual way, then updated the firmware to 13.2.  So far, so good.

Then added my MQTT (HA) server and voilà, HA discovered the device.

But here's the issue, the friendly name does not match. Like this page state: https://tasmota.github.io/docs/Home-Assistant/

Here's the config and the view from HA:
![image](https://github.com/arendst/Tasmota/assets/7669537/c386c603-2e03-4251-8bde-401bf87f4022)
![image](https://github.com/arendst/Tasmota/assets/7669537/7698d0a1-6315-4100-8b86-cb359f0aaa90)

I think I saw this issue before with many relays.

So then I tried so many things!  I don't remember them all, but nothing worked.

Now, I deleted the device from HA side and I cannot add it anymore!

If you have some ideas, let me know! 

Just tried restarting HA... nope, not working.

Thanks!

Simon

**Comment by barbudor** [comment]
Set `Weblog 4`and restart with `Restart 1`
Look at the console and you should have a log similar to:
```
22:38:18.445 MQT: tasmota/discovery/18FE34A6D683/config = {"ip":"192.168.168.195","dn":"MyRelayBoard","fn":["Relay1","Relay2","Relay3","Relay4","Relay5","Relay6","Relay6","Relay8"],"hn":"nodemcu","mac":"18FE34A6D683","md":"Athom R08","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.2.0.2","t":"nodemcu","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":1,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1} (retained)
22:38:18.457 MQT: tasmota/discovery/18FE34A6D683/sensors = {"sn":{"Time":"2023-11-07T22:38:18+01:00"},"ver":1} (retained)
```

On the 1st line, you should see `"dn":"yourdevicename"` and `"fn":[ ... your Friendly names ...]"`
If this is ok, the problem is on your HA side and should be addressed on HA forums
Do you have the latest version of [hatasmota](https://github.com/emontnemery/hatasmota) ?

**Reply by SimonFili** [reply]
Ok I got this:

```
23:17:41.421 MQT: tasmota/discovery/8CAAB5152EFD/config = {"ip":"192.168.10.114","dn":"Relay_Switch_X8","fn":["ATHOM_Rx8_1","ATHOM_Rx8_2","ATHOM_Rx8_3","ATHOM_Rx8_4","ATHOM_Rx8_5","ATHOM_Rx8_6","ATHOM_Rx8_7","ATHOM_Rx8_8"],"hn":"tasmota-152EFD-3837","mac":"8CAAB5152EFD","md":"Relay_Switch_X8","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.2.0","t":"tasmota_152EFD","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1} (retained)
23:17:41.432 MQT: tasmota/discovery/8CAAB5152EFD/sensors = {"sn":{"Time":"2023-11-07T23:17:41"},"ver":1} (retained)

```
And HA did not add it. 

As for hatasmota, unsure what this is.  I have HAOS installed in a VM, all updated.  I have around 20 other Tasmota devices connected.

**Reply by barbudor** [reply]
HaTasmota is a HA module/plug-in that process the above message to create the entities in HA
Above is correct so kit Tasmota fault

next step: HA forum

**Reply by rebel7580** [reply]
I'm SWAGing here, but did you have this set up originally with old Tasmota sw that only did the first 4 relays? If so, those first four names have been 'remembered" by HA. When you upgraded and got the last four to show up (and be discovered) HA added these in but used the "newer format" for entity friendly names of {device_name friendly_name}, That's likely why you see the device name show up on the last four. (truncated strings. Look at one of the entities directly to see the full name.)

See the first couple of lines in this post: https://community.home-assistant.io/t/why-does-mqtt-discovery-add-device-name-to-entity-name/637396/2?u=rebel7580

You will probably need to delete the device in HA (publish to topic tasmota/discovery/8CAAB5152EFD/config with null payload is one way. Then restart the Tasmota device again to re-discover it.
